core/src/main/java/brooklyn/location/access/PortForwardManagerImpl.java

```diff
@@ -181,6 +187,15 @@ protected void associateImpl(String publicIpId, HostAndPort publicEndpoint, Loca
         onChanged();
     }
 
+    private void emitAssociationCreatedEvent(String publicIpId, HostAndPort publicEndpoint, Location location, int privatePort) {
+        AssociationMetadata metadata = new AssociationMetadata(publicIpId, publicEndpoint, location, privatePort);
+        for (Map.Entry<AssociationListener, Predicate<? super AssociationMetadata>> entry : associationListeners.entrySet()) {
+            if (entry.getValue().apply(metadata)) {
+                entry.getKey().onAssociationCreated(metadata);
```

Comment:
Should wrap this in try-catch block, to log.warn any exception. An exception in one listener shouldn't cause the other listeners to not be called. However, in the catch block first do `Exceptions.propagateIfFatal(e)`.

Same goes for the call to `onAssociationDeleted(metadata)`.